install_config/default_imagestreams_templates.adoc

```diff
@@ -0,0 +1,420 @@
+- link:../../architecture/core_concepts/deployments.html#deployments-and-deployment-configurations[Deployment configurations]
+endif::[]
+to deploy the application image after it is built.
+ifdef::openshift-enterprise[]
+- link:../architecture/core_concepts/pods_and_services.html#services[Services]
+endif::[]
+ifdef::openshift-origin[]
+- link:../../architecture/core_concepts/pods_and_services.html#services[Services]
+endif::[]
+to provide load balancing for the application
+ifdef::openshift-enterprise[]
+link:../architecture/core_concepts/pods_and_services.html#pods[pods].
+endif::[]
+ifdef::openshift-origin[]
+link:../../architecture/core_concepts/pods_and_services.html#pods[pods].
+endif::[]
+ifdef::openshift-enterprise[]
+- link:../architecture/core_concepts/routes.html[Routes]
+endif::[]
+ifdef::openshift-origin[]
+- link:../../architecture/core_concepts/routes.html[Routes]
+endif::[]
+to provide external access to the application.
+
+Some of the templates also define a database deployment and service so the
+application can perform database operations.
+
+[NOTE]
+====
+The templates which define a database use ephemeral storage for the database
+content. These templates should be used for demonstration purposes only as all
+database data will be lost if the database pod restarts for any reason.
+====
+
+After creating the templates, users are able to easily instantiate full
+applications using the various language images provided with OpenShift. They can
+also customize the template parameters during instantiation so that it builds
+source from their own repository rather than the sample repository, so this
+provides a simple starting point for building new applications.
+
+To create the core InstantApp templates:
+
+----
+$ oc create -f $QSTEMPLATES -n openshift
+----
+
+ifdef::openshift-enterprise[]
+There is also a set of templates for creating applications using various xPaaS
+Middleware products (link:../using_images/xpaas_images/eap.html[*JBoss EAP*],
+link:../using_images/xpaas_images/jws.html[*JBoss JWS*],
+link:../using_images/xpaas_images/a_mq.html[*JBoss A-MQ*],
+link:../using_images/xpaas_images/fuse.html[*JBoss Fuse Integration Services*],
+link:../using_images/xpaas_images/decision_server.html[*Decision Server*], and
+link:../using_images/xpaas_images/data_grid.html[*JBoss Data Grid*]), which can
+be registered by running:
+
+----
+$ oc create -f $XPAASTEMPLATES -n openshift
+----
+
+[NOTE]
+====
+The xPaaS Middleware templates require the
+link:../install_config/default_imagestreams_templates.html#creating-image-streams-for-xpaas-middleware-images[xPaaS
+Middleware image streams], which in turn require the relevant xPaaS Middleware
+subscriptions.
+====
+
+[NOTE]
+====
+The templates which define a database use ephemeral storage for the database
+content. These templates should be used for demonstration purposes only as all
+database data will be lost if the database pod restarts for any reason.
+====
+endif::[]
+
```

Comment:
that's fair, but we should still be talking about those other templates, even if we (incorrectly) refer to them as instantapps and come back later to fix the naming.